public/app/plugins/panel/timeseries/utils.ts

```diff
@@ -123,3 +136,41 @@ export function getTimezones(timezones: string[] | undefined, defaultTimezone: s
   }
   return timezones.map((v) => (v?.length ? v : defaultTimezone));
 }
+
+export function regenerateLinksSupplier(
+  alignedDataFrame: DataFrame,
+  frames: DataFrame[],
+  replaceVariables: InterpolateFunction,
+  timeZone: string
+): DataFrame {
+  alignedDataFrame.fields.forEach((field) => {
+    const frameIndex = field.state?.origin?.frameIndex;
+
+    if (frameIndex === undefined) {
+      return;
+    }
+
+    const frame = frames[frameIndex];
+    const tempFrame = cloneDeep(alignedDataFrame);
+
+    /* check if field has sortedVector values
+      if it does, sort all string fields in the original frame by the order array already used for the field
+      otherwise just attach the fields to the temporary frame used to get the links
+    */
+    for (const frameField of frame.fields) {
+      if (frameField.type === FieldType.string) {
+        if (field.values instanceof SortedVector) {
+          const clonedFrameField = cloneDeep(frameField);
+          clonedFrameField.values = new SortedVector(frameField.values, field.values.getOrderArray());
+          tempFrame.fields.push(clonedFrameField);
+        } else {
+          tempFrame.fields.push(frameField);
+        }
+      }
+    }
+
+    field.getLinks = getLinksSupplier(tempFrame, field, field.state!.scopedVars!, replaceVariables, timeZone);
+  });
```

Comment:
This looks a bit scary from a performance standpoint. Does this run when data updates or when the user tries to view the datalinks?

Reply:
agreed. i cannot review for a few days (out of office)

Reply:
This runs every time the data updates. I took out the `cloneDeep` calls